streaming/base/dataset.py

```diff
@@ -155,15 +155,22 @@ def __init__(self,
             raise ValueError(
                 'Parameter ``download_timeout`` (in seconds) must be greater than zero')
 
+        # Placeholder for _resume_shm, a shared memory object where load_state_dict() saves its
+        # data to be picked up by __iter__().
+        self._resume_shm = None
+
+        # Partition state.
+        self._partition_state = None
+
         # Seed is set below.
-        world = World()
+        self.world = World()
```

Comment:
Ehhh, it's dangerous to save the world state as a member of StreamingDataset

(i.e., the results will be correct or incorrect depending on whether you are in a worker or rank process -- someone may be tempted in the future to reuse self.world from some code descending from `__iter__` and silently get the wrong state)

Reply:
As discussed offline, instantiating a new world state during termination stage. Addressed in https://github.com/mosaicml/streaming/pull/107/commits/44e4a951687ec2bf2a58c35e3a164f77562af7be.